storage/series/reader.go

```diff
@@ -179,11 +177,17 @@ func (r Reader) fetchBlocksWithBlocksMapAndBuffer(
 	seriesBlocks block.DatabaseSeriesBlocks,
 	seriesBuffer databaseBuffer,
 ) ([]block.FetchBlockResult, error) {
-	res := make([]block.FetchBlockResult, 0, len(starts))
-
-	var clonedID ts.ID
-	cachePolicy := r.opts.CachePolicy()
-
+	var (
+		// TODO(r): pool these results arrays
+		res         = make([]block.FetchBlockResult, 0, len(starts))
+		cachePolicy = r.opts.CachePolicy()
+		// NB(r): Always use nil for OnRetrieveBlock so we don't cache the
```

Comment:
Comment makes sense to me, this method is only used during peer bootstrapping then?

Reply:
Correct, yeah fetch blocks is only for the peer bootstrapping case.